src/Downloading/UnixDownloadAndExtract.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Php\Pie\Downloading;
+
+use Composer\Util\AuthHelper;
+use GuzzleHttp\Psr7\Request;
+use Php\Pie\DependencyResolver\Package;
+use Psr\Http\Message\RequestInterface;
+use RuntimeException;
+
+use function file_exists;
+use function mkdir;
+use function sprintf;
+use function sys_get_temp_dir;
+use function uniqid;
+
+/** @internal This is not public API for PIE, so should not be depended upon unless you accept the risk of BC breaks */
+final class UnixDownloadAndExtract implements DownloadAndExtract
+{
+    public function __construct(
+        private readonly DownloadZip $downloadZip,
+        private readonly ExtractZip $extractZip,
+        private readonly AuthHelper $authHelper,
+    ) {
+    }
+
+    public function __invoke(Package $package): DownloadedPackage
+    {
+        // @todo extract to a static util
+        $localTempPath = sys_get_temp_dir() . '/' . uniqid('pie_downloader_', true);
```

Comment:
Please don’t `uniqid()` in newly written greenfield code. See: https://wiki.php.net/rfc/deprecations_php_8_4#deprecate_uniqid

My suggestion would be either `sprintf('pie_downloader_%s', bin2hex(random_bytes($n)))` (with `$n` at least 8 and ideally 16) or [`tempnam()`](https://www.php.net/manual/en/function.tempnam.php).

Reply:
@TimWolla Thanks for pointing out the RFC, however the RFC is not yet accepted, and IMO this is an acceptable use of `uniqid`; strong randomness is absolutely not required here, a time-based randomness is ideal. PIE is a tool run by a human, or in a CI pipeline, so the risk of two separate PIE processes generating the same path is extremely low.